src/importlinter/contracts/layers.py

```diff
@@ -29,18 +33,38 @@ class Layer:
 
 class LayerField(fields.Field):
     def parse(self, raw_data: str | list) -> Layer:
-        module_tails = set()
         raw_string = fields.StringField().parse(raw_data)
-        raw_items = [item.strip() for item in raw_string.split("|")]
-        for raw_item in raw_items:
-            if raw_item.startswith("(") and raw_item.endswith(")"):
-                module_tail = raw_item[1:-1]
+        raw_module_tails, independent = self._split_raw_module_tails(raw_string)
+        module_tails = set()
+        for raw_module_tail in raw_module_tails:
+            if raw_module_tail.startswith("(") and raw_module_tail.endswith(")"):
+                module_tail = raw_module_tail[1:-1]
                 is_optional = True
             else:
-                module_tail = raw_item
+                module_tail = raw_module_tail
                 is_optional = False
             module_tails.add(ModuleTail(name=module_tail, is_optional=is_optional))
-        return Layer(module_tails, independent=True)
+        return Layer(module_tails, independent=independent)
+
+    @staticmethod
+    def _split_raw_module_tails(raw_string: str) -> tuple[list[str], bool]:
+        if (
+            _INDEPENDENT_LAYER_DELIMITER in raw_string
+            and _NON_INDEPENDENT_LAYER_DELIMITER in raw_string
+        ):
+            raise fields.ValidationError("Layer cannot be both independent and non-independent")
```

Comment:
Would be clearer to say "Layer cannot have a mixture of independent and non-independent elements."?

Also, can the error message have a full stop at the end of the sentence?